arangod/RestHandler/RestDocumentHandler.cpp

```diff
@@ -784,9 +788,8 @@ RestStatus RestDocumentHandler::removeDocument() {
 
   _activeTrx = createTransaction(cname, AccessMode::Type::WRITE, opOptions,
                                  std::move(trxOpts));
-  if (suffixes.size() == 2 || !search.isArray()) {
-    _activeTrx->addHint(transaction::Hints::Hint::SINGLE_OPERATION);
-  }
+
+  addTransactionHints(cname, isMultiple, false);
```

Comment:
The case `suffixes.size() == 2` is not optimized anymore, but it should be:
```suggestion
  addTransactionHints(cname, isMultiple || suffixes.size() != 2, false);
```

Reply:
At actually use, because the suffixes == 2 case will Produce an Object, not an array, and isMultiple is only true if we have an array in the search clause.
So we hand in false, false here, which will trigger the SINGLE_OPERATION hint